base/reinterpretarray.jl

```diff
@@ -44,6 +44,7 @@ function size(a::ReinterpretArray{T,N,S} where {N}) where {T,S}
     tuple(size1, tail(psize)...)
 end
 
+elsize(::Type{<:ReinterpretArray{T}}) where {T} = sizeof(T)
```

Comment:
This doesn't seem totally safe --- it seems like the parent array type would need to be involved somehow.

Reply:
There's no relation imposed by this code between the memory layout of a reinterpret array and its parent. However, it does impose that `T` is a known `isbits` type, which has a known memory size (although not stride) equal to `sizeof(T)`

Reply:
I think Jeff is right.  Am I correct in thinking about the three cases here?

* `sizeof(eltype(R)) < sizeof(eltype(R.parent))`: the `elsize` of the parent should appear in `ReshapedArray`'s second stride.
* `sizeof(eltype(R)) == sizeof(eltype(R.parent))`: `elsize` and `strides` should match
* `sizeof(eltype(R)) > sizeof(eltype(R.parent))`: The reshaped array is only a strided array if `sizeof(eltype(R.parent)) == elsize(R.parent)`.  We cannot pass this object to BLAS otherwise, since BLAS will assume that the bits are contiguous.

Reply:
Note that I'm going to follow this up with the `strides` portion of #26072, which will include part of that.  I'm not sure how to handle that third bullet point, though.

Reply:
I guess we should maybe merge this now anyway, since it's no worse than it was before.

Reply:
It's less of a problem now that #26393 is merged — only reinterprets of `DenseArrays` (and contiguous views thereof) are considered strided now.  We'll have to be very careful about how `ReinterpretArray`s opt-in to being considered strided in the future.  I think we'll just need to demand that `elsize(parent(R)) == sizeof(eltype(parent(R)))` and `stride(parent(R), 1) == 1`.  In all other cases, this cannot be correct.